### ISSUE TITLE
ACEX Headless always rebalances units

### ISSUE THREAD

**Issue by specop0** [acemod/ACEX#90]
**Arma 3 Version:** `1.80` (stable)
**CBA Version:** `3.5` (stable)
**ACE3 Version:** `3.11.0` (stable)
**ACEX Version:** `3.2.0` (stable)

**Mods:**
```
- CBA_A3
- ace
- acex
- a bunch of others
```

**Description:**
- ACEX Headless always rebalances the units, even if disabled

https://github.com/acemod/ACEX/blob/master/addons/headless/XEH_preInit.sqf#L13

https://github.com/acemod/ACEX/blob/master/addons/headless/functions/fnc_handleConnectHC.sqf#L32

the QGVAR(enabled) is only checked in
- https://github.com/acemod/ACEX/blob/master/addons/headless/functions/fnc_handleDisconnect.sqf#L50
- https://github.com/acemod/ACEX/blob/master/addons/headless/XEH_postInit.sqf#L5-L8

Should be checked in handleConnectHC or used as param?

**Steps to reproduce:**

1. Start attached mission with headless client setup [missions.zip](https://github.com/acemod/ACEX/files/1544686/missions.zip)
2. Groups are always Rebalanced / Rebalance is always true

**Where did the issue occur?**
- Multiplayer

**Placed Modules:**
- diverse modules placed via serverconfig
- ACEX Headless Modul if placed in mission
-- Enabled: Yes or No
-- Delay: 15
-- End Mission: Delayed (60s)
-- Log: Disabled
-- via serverconfig: Enabled: No, Log: Enabled (force)

Why do we use this setup?
We use a server side module to let ACEX headless end the mission, because the HC always connects. But each mission must manually enable ACEX headless rebalancing, because not every mission maker can make a mission which supports a HC correctly.
see https://github.com/acemod/ACEX/issues/24

**RPT log file:**
- https://pastebin.com/GFHMJq1E

**Comment by jonpas** [comment]
It is checked on start https://github.com/acemod/ACEX/blob/74861b4cabd26b8aa0e5966c182d666d8da0c41f/addons/headless/XEH_postInit.sqf#L5

Going by that RPT
```
13:43:33 [ACE] (common) INFO: SetSetting [acex_headless_endMission] Trying to set forced setting
13:43:33 [ACE] (common) INFO: SetSetting [acex_headless_log] Trying to set forced setting
```
You probably have the setting forced on the server or something.

**Comment by specop0** [comment]
If I understand the serverconfig correctly, only the endMission and log setting are forced - not the enabled flag, or am I missing something?
https://pastebin.com/8wyCVaUP

And yes, it is checked in the postInit, but not in the in the preInit:
https://github.com/acemod/ACEX/blob/master/addons/headless/XEH_preInit.sqf#L13

Where we register on the headlessClientJoind event and trigger a rebalance:
https://github.com/acemod/ACEX/blob/master/addons/headless/functions/fnc_handleConnectHC.sqf#L32

So if I'm understanding correctly, all clients are rebelanced if a HC joins for the first time. But units spawned in a mission (via script or zeus) are not rebelanced.

**Comment by jonpas** [comment]
Hmh, I could bet it doesn't do anything when disabled.

**Comment by specop0** [comment]
Can I somehow check the actual setting in-game of the enabled flag?

I would at first guess something like getting a variable is correct:
```str (missionNamespace getVariable ["acex_headless_enabled", "NONE"]) remoteExec ["hint"];```

The result from the server is "false" for the mission without module/with server setting only and module with disabled rebalancing.

But a mission maker can set such variable, so I'm not sure if this is really the case (if no magic happens from CBA if this variable is set).

**Comment by jonpas** [comment]
It's just a global variable, so you can just read it directly by putting it into the watch console. And enable [CBA Target Debugging](https://github.com/CBATeam/CBA_A3/wiki/Target-Debugging) to read it from the server.

**Comment by PabstMirror** [comment]
Can also quickly check server val by `Server Exec` this:
`(format ["Server Val: %1", acex_headless_enabled]) remoteExec ["hint"]`

**Comment by specop0** [comment]
@PabstMirror which is the same I already did. But now I know a better debug window .. hehe

![debug-window.png](https://user-images.githubusercontent.com/11978783/33797547-632d35e6-dd0a-11e7-9465-5324b5ae429e.png)

[missions_v04.zip](https://github.com/acemod/ACEX/files/1544851/missions_v04.zip)

**Comment by jonpas** [comment]
I see where this bug came from, we still want to track HCs connecting and disconnecting, so we can do the End Mission feature correctly.

So like we do here:
https://github.com/acemod/ACEX/blob/e6b9e2b3cf8e4235868baf1a236504e21120fda0/addons/headless/functions/fnc_handleDisconnect.sqf#L49-L50

We should do the same before this:
https://github.com/acemod/ACEX/blob/e6b9e2b3cf8e4235868baf1a236504e21120fda0/addons/headless/functions/fnc_handleConnectHC.sqf#L31-L32

**Comment by specop0** [comment]
Yeah, this is what I meant...
https://github.com/specop0/ACEX/commit/03e11c4f79bb3108f1515ed1d4a5891aeabb0675

**Comment by jonpas** [comment]
Aaah, I should've read your original report better, sorry. :(

Can you make a PR for that? If you can't test, that's fine, but would be great if you could.

**Comment by specop0** [comment]
I cannot test the change because I don't have the build environment set up...PR is pending (I hope everything correctly...)

**Comment by jonpas** [comment]
Fixed by #91.
